### ISSUE TITLE
Code in the Documentation is resulting in Warning Message

### ISSUE THREAD

**Issue by worldpeaceaspirer** [tensorflow/tensorflow#50190]
## URL(s) with the issue:

Please provide a link to the documentation entry, for example:
https://www.tensorflow.org/guide/keras/writing_a_training_loop_from_scratch#using_the_gradienttape_a_first_end-to-end_example

## Description of issue (what needs changing): 
Need to modify the code such that Warning goes away.

Please find the [Github Gist](https://colab.research.google.com/gist/worldpeaceaspirer/73e9f295a64cb513d6b7cd3aefdc2c38/warning_in_the_tutorial.ipynb) that demonstrates the Warning.

**Comment by UsharaniPagadala** [comment]
@worldpeaceaspirer

In order to expedite the trouble-shooting process, Could you please elaborate your  issue  with complete details. And what's need to be modified.Thanks

**Comment by MarkDaoust** [comment]
I see the warning in your gist, but not in the tensorflow.org page.

So I think the page is working as intended.

Or am I misunderstanding something?

**Comment by ymodak** [comment]
I think you get the warning because you are using `softmax` activation function in the output layer whereas original example uses none. The warning goes away by removing `softmax` activation function.
```python
outputs = tf.keras.layers.Dense(10, name="predictions")(x2)
```

**Comment by worldpeaceaspirer** [comment]
@MarkDaoust,
Warning is not appearing in the documentation and is appearing when we execute the code and that's the reason for my confusion. The in the documentation should be Error-Free and Warning-Free.

Do you mean to say the Warning is expected? If it is so, we can specify the same in the page itself, to avoid confusion. If the warning is not intended, code should be modified to remove that warning.

Can you please reopen this issue

**Comment by worldpeaceaspirer** [comment]
> I think you get the warning because you are using `softmax` activation function in the output layer whereas original example uses none. The warning goes away by removing `softmax` activation function.
> 
> ```python
> outputs = tf.keras.layers.Dense(10, name="predictions")(x2)
> ```

@ymodak,
It is not technically correct to remove **`Softmax Activation`** from the **`Output Layer`** in case of **`Multi-Class Classification`** right?

**Comment by MarkDaoust** [comment]
@worldpeaceaspirer It depnds.

The last layer output, the loss function, and the metrics all need to be set to work in the same units (probability or logits).
